### ISSUE TITLE
RN android lost touches with E/unknown: Reactions: Got DOWN touch before receiving or CANCEL UP from last gesture

### ISSUE THREAD

**Issue by sneerin** [facebook/react-native#17073]
### Is this a bug report?

yes, bug
### Environment

  Environment:
   Environment:
  OS:  macOS High Sierra 10.13.1
  Node:  8.4.0
  Yarn:  0.17.8
  npm:  5.3.0
  Watchman:  4.9.0
  Xcode:  Xcode 9.1 Build version 9B55
  Android Studio:  2.3 AI-162.4069837

Packages: (wanted => installed)
  react: 16.0.0-alpha.12 => 16.0.0-alpha.12
  react-native: 0.48.4 => 0.51.0
  Target Platform: Android (7)

### Steps to Reproduce

Any overlay cause touch not working for all native and rn views with error:
E/unknown: Reactions: Got DOWN touch before receiving or CANCEL UP from last gesture
JSTouchDispatcher.java causing this handler message

### Expected Behavior

touches remain working 

(Write what you thought would happen.)
something deep in the android rn implementation happened.
### Actual Behavior

touches doesn't work on android

### Reproducible Demo

https://www.npmjs.com/package/react-native-loading-spinner-overlay use this component, after it loads several times the app become unresponsible, but not only this one.
UPDT: reproduced in both dev and prod modes.
UPDT: the same issue is happening when

>  import { ProgressDialog } from 'react-native-simple-dialogs';

and then showing it several times really frequently.

**Comment by ChengYuHin99** [comment]
same issue for me too.

**Comment by binlaniua** [comment]
the same to me.

**Comment by sneerin** [comment]
before I was thinking it's only dev related, but right now in prod mode it's not working, it's also blocked touches for also native ui, actually it makes all application not working.

**Comment by RajeshBatth** [comment]
Even i'm getting this issue in v0.51.0, any update on this?

**Comment by daimashenjing** [comment]
same issue to me too.

**Comment by daimashenjing** [comment]
![image](https://user-images.githubusercontent.com/8486201/35256420-c53b89d0-002e-11e8-91d9-32ae7bf0a1cc.png)

**Comment by juumixx** [comment]
I had the same issue. I think it was caused by the Spinner getting initialized but not being drawn or attached because it was already hidden again when the view becomes visible.
I solved it by setting the visibility of the spinner through a private observable:

```typescript
export class BaseViewContainer extends React.Component<Props, State> { 
    private dispose: any;
    @observable private showSpinner: boolean;

    componentDidMount() { 
        this.dispose = autorun(() => { 
            this.showSpinner = this.props.spinnerMessage !== undefined; 
            }
        );
    }

    componentWillUnmount() {
        this.dispose();
    }

    render() {
        return (
            <View /*...*/>
                 <Spinner visible={this.showSpinner}>  
                //...
                </Spinner>
                {this.props.children}
            </View>
        );
    }
}
```

Can you provide some code examples of your problems? Maybe this helps finding the real issue.

**Comment by daimashenjing** [comment]
测试了一下果然如@juumixx 所说一样 当一个控件被显示，显示操作还没执行完成，控件没有完全显示出来，这时设置隐藏就会导致这个bug,卡住整个app, 强制结束重启后才能继续使用. 
我在关闭对话框时加入InteractionManager就没有出现这个问题
![image](https://user-images.githubusercontent.com/8486201/35313285-7f9e8f54-00fa-11e8-8baf-ca84fb542d0e.png)

**Comment by juumixx** [comment]
Ok actually it was still broken. Even the InteractionManager.runAfterInteractions() hack didn't work.
Found a workaround by blocking all the actions during a navigation change. So maybe it is also related to react-navigation.

**Comment by FlorianBergmann** [comment]
I have the same issue but I am not using the Spinner library. Instead I am using the ActivityIndicator but even after removing it, I still get this error. 

So far I can reproduce it by starting the app and quickly pressing the button that opens the navigation drawer (my app is for Android) of react-navigation. This usually triggers the error and I can only get rid of it by killing the app.

**Comment by daimashenjing** [comment]
我在较差的手机上还是出现了  我怀疑和 modal 有关系

**Comment by ptvandi** [comment]
I am still receiving these errors in 0.53.3.

**Comment by Qiu800820** [comment]
I switched to dev mode, this issues never appeared

**Comment by mateoc10** [comment]
@FlorianBergmann could you solve it? I am having the same issue. I'm going crazy!

**Comment by vadermemo** [comment]
Same issue here, even with 

    "react": "16.3.1",
    "react-native": "0.55.4",

Just on Android! not on iOS.

**Comment by lin-xin** [comment]
+1

**Comment by IndujaVJ** [comment]
@Qiu800820 Do you still have the issue?

**Comment by Qiu800820** [comment]
@IndujaVJ Model show or hide action I use setState(newState, callback)，Never again this issue

**Comment by daimashenjing** [comment]
Got DOWN touch before receiving UP or CANCEL from last gesture
在错误（红屏）界面点击返回键会触发这个bug 目前我是100%触发

**Comment by daimashenjing** [comment]
android
"react-native": "0.55.4",
error page (Red screen)  Click back（entity key） 100%
在错误（红屏）界面点击返回键会触发这个bug目前我是100％触发
```
07-05 11:29:21.111 11925-11925/com.ar3 E/unknown:ReactNative: Got DOWN touch before receiving UP or CANCEL from last gesture
07-05 11:29:22.182 11925-11925/com.ar3 E/unknown:ReactNative: Got DOWN touch before receiving UP or CANCEL from last gesture
07-05 11:29:22.732 11925-11925/com.ar3 E/unknown:ReactNative: Got DOWN touch before receiving UP or CANCEL from last gesture
07-05 11:29:25.631 11925-11925/com.ar3 E/unknown:ReactNative: Got DOWN touch before receiving UP or CANCEL from last gesture

```

**Comment by FlorianBergmann** [comment]
We were able to get rid of it as well. Our issue seemed to relate to our implementation of Dialogs with Modal view. We made several changes and I am not quite sure what fixed it in the end. Originally we handled the visible state of the Modal from outside of our dialogs using the state of the parent. In our current implementation we instead have the visible state inside the dialog and change if from the parent by saving a reference and calling a method. 

@mateoc10 sorry for the late reply.
